include/net/nrf_cloud_coap.h

```diff
@@ -301,23 +301,24 @@ int nrf_cloud_coap_fota_job_update(const char *const job_id,
 	const enum nrf_cloud_fota_status status, const char * const details);
 
 /**
- * @brief Query the device's shadow delta.
- *
- * If there is no delta, the return value will be 0 and the length of the string stored
- * in buf will be 0.
+ * @brief Query the device's delta or desired shadow section.
  *
  * @param[in,out] buf     Pointer to memory in which to receive the delta.
  * @param[in,out] buf_len Size of buffer, will be set to the incoming length.
  * @param[in]     delta   True to request only changes in the shadow, if any; otherwise,
- *                        all of desired part.
+ *                        all of the desired section.
+ *                        If there is no data in the specified section, the return value will be 0
+ *                        and the length of the string stored in buf will be 0.
```

Comment:
nit: perhaps generalize the language a bit to also cover cbor as content length where we don't use strings?